File: sdks/php/src/Types/QueueMessage.php
<?php

declare(strict_types=1);

namespace Synap\SDK\Types;

/**
 * Represents a message in a queue
 */
readonly class QueueMessage
{
    public function __construct(
        public string $id,
        public mixed $payload,
        public int $priority = 0,
        public int $retries = 0,
        public int $maxRetries = 3,
        public int $timestamp = 0
    ) {
    }

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'payload' => $this->payload,
            'priority' => $this->priority,
            'retries' => $this->retries,
            'max_retries' => $this->maxRetries,
            'timestamp' => $this->timestamp,
        ];
    }
}

File: sdks/csharp/src/Synap.SDK/Types/QueueMessage.cs
using System.Text.Json;

namespace Synap.SDK.Types;

/// <summary>
/// Represents a message from a queue.
/// </summary>
public sealed record QueueMessage
{
    /// <summary>
    /// Gets the message ID.
    /// </summary>
    public required string Id { get; init; }

    /// <summary>
    /// Gets the message payload.
    /// </summary>
    public required JsonElement? Payload { get; init; }

    /// <summary>
    /// Gets the message priority (0-9, higher is more important).
    /// </summary>
    public int Priority { get; init; }

    /// <summary>
    /// Gets the number of times this message has been retried.
    /// </summary>
    public int Retries { get; init; }

    /// <summary>
    /// Gets the maximum number of retries allowed.
    /// </summary>
    public int MaxRetries { get; init; } = 3;

    /// <summary>
    /// Gets the message timestamp (Unix timestamp in seconds).
    /// </summary>
    public long Timestamp { get; init; }
}

